### DISCUSSION TITLE
CORS error with CefCustomScheme

### DISCUSSION THREAD

**Discussion by ray007** [cefsharp/CefSharp#4097] [General]
I have an web-application loaded from the local filesystem with a `CefCustomScheme` handler with address `http://myapp.localhost/`.

After upgrading to version 101 (from version 85), requests to another `CefCustomScheme` registered to `http://prjdata.localhost/` suddenly fail, requests to another scheme handler registered with a custom SchemeName instead still seem to work.

Any hints which options I need to supply when registering the scheme handler (or settings switches) to make things just work?

**Comment by amaitland** [comment]
I'd suggest moving the discussion to https://magpcss.org/ceforum/viewforum.php?f=6 (post a link back here for reference).

**Comment by ray007** [comment]
Funny discovery: changing my code from using `XmlHttpRequest` (via jQuery ajax) to using `fetch` (with `mode = 'no-cors'` and adding `IsFetchEnabled = true` to the scheme handler registration), things now seem to work too.
Will do more testing, but if things work out I'll skip getting involved on the CEF forum.

**Comment by ray007** [comment]
Update: requests with POST data arrive in my custom scheme handler, custom http headers set in the js app do not. Neither with http nor with custom scheme.

**Reply by amaitland** [reply]
My only suggestion would be to try using a domain other than localhost. Otherwise ask on ceforum.

**Reply by ray007** [reply]
> My only suggestion would be to try using a domain other than localhost. Otherwise ask on ceforum.

I'm using the `.localhost` domain since it's a reserved tld according to rfc 2606.

Since using query parameters instead of custom headers works well enough for now I'm good, thanks for your help.

I don't suppose there's a more efficient method for binary data transfer between C#-layer and js-app than a scheme handler, is there?

**Reply by amaitland** [reply]
> I'm using the .localhost domain since it's a reserved tld according to rfc 2606.

Whilst that is true my concern would be that chromium treats localhost as a special case.

> I don't suppose there's a more efficient method for binary data transfer between C#-layer and js-app than a scheme handler, is there?

Not yet.
